### PR TITLE
Fix clang warnings and other minor issues

### PR DESCRIPTION
**Pull request by cyyever**
Fix various clang warnings.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @EikanWang

### CODE REVIEW

File: aten/src/ATen/native/quantized/cpu/conv_serialization.h [hunk 1]
```diff
@@ -123,8 +123,6 @@ ConvParamsSerializationTypeV3 parse_conv_serialized_state(c10::IValue v) {
     torch::List<at::Tensor> dilation_x_kSpatialDim = elements[4].toTensorList();
     at::Tensor groups = elements[5].toTensor();
 
-    std::vector<c10::optional<at::Tensor>> optional;
-
     std::vector<int64_t> config_vals;
```
Comment (Skylion007) [inline aten/src/ATen/native/quantized/cpu/conv_serialization.h:126]: Hmm, seems like config_vales should have .reserve() called on it with the various of sizes of stuff being added to it.

File: aten/src/ATen/native/RNN.cpp [hunk 2]
```diff
@@ -405,8 +405,6 @@ struct QuantizedCellParamsDynamic : public CellParamsBase {
   CellParamsSerializationType __getstate__() const override {
     // Boxed dispatch nonsense
```
Comment (Skylion007) [inline aten/src/ATen/native/RNN.cpp:406]: Is this the boxed dispatch nonsense? If so should the comment be removed?

File: aten/src/ATen/native/quantized/cpu/conv_serialization.h (hunk 1 above)
Comment (cyyever) [reply aten/src/ATen/native/quantized/cpu/conv_serialization.h:126]: It is hard to reserve() in some cases

Comment (Skylion007) [reply aten/src/ATen/native/quantized/cpu/conv_serialization.h:126]: Should just be
`.reserve(stride_x_kSpatialDim.size() + padding_x_kSpatialDim.size() + dilation_x_kSpatialDim.size() + kSpatialDim + 3)`
right?

**Review by Skylion007 — APPROVED** [review]

File: aten/src/ATen/native/quantized/cpu/conv_serialization.h (hunk 1 above)
Comment (cyyever) [reply aten/src/ATen/native/quantized/cpu/conv_serialization.h:126]: Thanks for pointing it out, I have added.

**Pull request closed without merging**